app/scripts/controllers/network/network-controller.test.js

```diff
@@ -50,224 +69,4106 @@ const PRE_1559_BLOCK = {
+                });
+                expect(
+                  controller.store.getState().networkDetails.EIPS['1559'],
+                ).toBeUndefined();
+              },
+            );
+          });
+        });
+      });
+
+      describe('if the request for net_version responds with an error', () => {
+        it('resets the network status to "loading"', async () => {
+          await withController(
+            {
+              state: {
+                provider: {
+                  type: 'rpc',
+                  rpcUrl: 'https://mock-rpc-url',
+                  chainId: '0x1337',
+                },
+              },
+            },
+            async ({ controller, network }) => {
+              network.mockEssentialRpcCalls({
+                net_version: [
+                  {
+                    response: {
+                      result: '42',
+                    },
+                  },
+                  {
+                    response: {
+                      error: 'oops',
+                    },
+                  },
+                ],
+              });
+              await controller.initializeProvider();
+              expect(controller.store.getState().network).toStrictEqual('42');
+
+              await waitForStateChanges({
+                controller,
+                propertyPath: ['network'],
+                operation: async () => {
+                  await controller.lookupNetwork();
+                },
+              });
+              expect(controller.store.getState().network).toStrictEqual(
+                'loading',
+              );
+            },
+          );
+        });
+
+        it('removes from state whether the network supports EIP-1559', async () => {
+          await withController(
+            {
+              state: {
+                provider: {
+                  type: 'rpc',
+                  rpcUrl: 'https://mock-rpc-url',
+                  chainId: '0x1337',
+                },
+              },
+            },
+            async ({ controller, network }) => {
+              network.mockEssentialRpcCalls({
+                latestBlock: POST_1559_BLOCK,
+                net_version: [
+                  {
+                    response: {
+                      result: '42',
+                    },
+                  },
+                  {
+                    response: {
+                      error: 'oops',
+                    },
+                  },
+                ],
+              });
+              await controller.initializeProvider();
+              expect(controller.store.getState().networkDetails).toStrictEqual({
+                EIPS: {
+                  1559: true,
+                },
+              });
+
+              await waitForStateChanges({
+                controller,
+                propertyPath: ['networkDetails'],
+                operation: async () => {
+                  await controller.lookupNetwork();
+                },
+              });
```

Comment:
Nit: One more suggested newline:

```suggestion
              });

```